src/b-splines/cubic.jl

```diff
@@ -0,0 +1,241 @@
+        $symp = SimpleRatio(2,3) - sqr(1-$symfx) + SimpleRatio(1,2)*$sym_1m_fx_cub
+        $sympp = SimpleRatio(1,6)*$symfx_cub
+    end
+end
+
+# This assumes integral values ixm_d, ix_d, and ixp_d,
+# coefficients cm_d, c_d, and cp_d, and an array itp.coefs
+function index_gen{C<:Cubic,IT<:DimSpec{BSpline}}(::Type{BSpline{C}}, ::Type{IT}, N::Integer, offsets...)
+    if length(offsets) < N
+        d = length(offsets)+1
+        symm, sym, symp, sympp =  symbol("cm_",d), symbol("c_",d), symbol("cp_",d), symbol("cpp_",d)
+        return :($symm * $(index_gen(IT, N, offsets...,-1)) + $sym * $(index_gen(IT, N, offsets..., 0)) +
+                 $symp * $(index_gen(IT, N, offsets..., 1)) + $sympp * $(index_gen(IT, N, offsets..., 2)))
+    else
+        indices = [offsetsym(offsets[d], d) for d = 1:N]
+        return :(itp.coefs[$(indices...)])
+    end
+end
+
+# ------------ #
+# Prefiltering #
+# ------------ #
+
+function inner_system_diags{T,C<:Cubic}(::Type{T}, n::Int, ::Type{C})
+    du = fill(convert(T, SimpleRatio(1, 6)), n-1)
+    d = fill(convert(T, SimpleRatio(2, 3)), n)
+    dl = copy(du)
+    dl, d, du
+end
+
+"""
+`Flat` `OnGrid` amounts to setting `y_0'(x) = 0` at `x = 0`. Applying this gives
+a condition where:
+
+    -cm + cp = 0
+"""
+function prefiltering_system{T,TC}(::Type{T}, ::Type{TC}, n::Int,
+                                   ::Type{Cubic{Flat}}, ::Type{OnGrid})
+    dl, d, du = inner_system_diags(T, n, Cubic{Flat})
+    d[1] = d[end] = -one(T)  # set first and last diagonal to -1
+    du[1] = dl[end] = zero(T)  # set just off diagonal to 0
+
+    # Now Woodbury correction to set `[1, 3], [n, n-2] ==> 1`
+    specs = _build_woodbury_specs(T, n, (1, 3, one(T)), (n, n-2, one(T)))
+
+    Woodbury(lufact!(Tridiagonal(dl, d, du), Val{false}), specs...), zeros(TC, n)
+end
+
+# TODO: check to see if above method is the same as Quadratic
+
+"""
+`Flat` `OnCell` amounts to setting `y_0'(x) = 0` at `x = -1/2`. Applying this
+gives a condition where:
+
+    -9 cm + 11 c -3 cp + 1 cpp = 0
+"""
+function prefiltering_system{T,TC}(::Type{T}, ::Type{TC}, n::Int,
+                                   ::Type{Cubic{Flat}}, ::Type{OnCell})
+    dl, d, du = inner_system_diags(T,n,Cubic{Flat})
+    d[1] = d[end] = -9  # diagonal equal to -9
+    du[1] = dl[end] = 11  # off diagonal to 11
+
+    # now need Woodbury correction to set :
+    #    - [1, 3] and [n, n-2] ==> -3
+    #    - [1, 4] and [n, n-3] ==> 1
+    specs = _build_woodbury_specs(T, n,
+                                  (1, 3, T(-3)),
+                                  (n, n-2, T(-3)),
+                                  (1, 4, one(T)),
+                                  (n, n-3, one(T))
+                                  )
+
+    Woodbury(lufact!(Tridiagonal(dl, d, du), Val{false}), specs...), zeros(TC, n)
+end
+
+# NOTE: Line, OnGrid is implemented in quadratic.jl
+
+"""
+`Line` `OnCell` amounts to setting `y_0'(x) = 0` at `x = -1/2`. Applying this
```

Comment:
yep, definitely a typo. However, the condition on the coefficients below is correct (it does use `y''`)